src/main/java/com/minecolonies/coremod/entity/ai/util/StructureIterator.java

```diff
@@ -226,11 +226,18 @@ public Result advanceBlock(final AbstractEntityAIStructure abstractEntityAIStruc
             case CLEAR:
                 return advanceBlocks(this.theStructure::decrementBlock,
                   structureBlock -> structureBlock.doesStructureBlockEqualWorldBlock()
-                                      || structureBlock.worldBlock == Blocks.AIR);
+                                      || structureBlock.worldBlock == Blocks.AIR
+                                      || structureBlock.worldBlock instanceof FlowingFluidBlock);
             case BUILD:
                 return advanceBlocks(this.theStructure::incrementBlock, structureBlock -> doesStructureBlockEqualWorldBlock(structureBlock, abstractEntityAIStructure)
                                                                                          || structureBlock.block == Blocks.AIR
                                                                                          || !structureBlock.metadata.getMaterial().isSolid());
+            case FLUID_DETECT:
+            case FLUID_REMOVE:
```

Comment:
I think we only need fluid detect, fluid remove is independent of block progress. (It's basically while list is not empty, return same state).

Reply:
I didn't know the step would work without this, so i left it there, i will remove it then.

Reply:
you make it a normal state (like the completion state) with the if (list.hasElements ) return getState(). else return Completion (last state)

Reply:
I did this.